packages/visual-layer/src/layers/arc/arc.js

```diff
@@ -189,7 +78,57 @@ export default class ArcLayer extends BaseLayer {
      * @memberof ArcLayer
      */
     getPlotElementsFromSet (set) {
-        return selectElement(this.mount()).selectAll(this.elemType()).filter(d => set.indexOf(d.datum.uid) !== -1);
+        return selectElement(this.mount()).selectAll(this.elemType()).filter(d => set.indexOf(d.rowId) !== -1);
+    }
+
+    translatePoints (data) {
+        const { angle, color: colorAxis, radius: radiusAxis } = this.axes();
+        this._prevPieData = {};
+        const pieIndex = {};
+        const prevData = this._points[0] || [];
+        prevData.forEach((e, index) => {
+            this._prevPieData[e.rowId] = [e, index];
+            pieIndex[e.index] = e;
+        });
+        const points = [];
```

Comment:
Declare consts and let first, then perform other processes, enhances readability and makes code cleaner